src/v/kafka/server/connection_context.cc

```diff
@@ -298,6 +313,20 @@ ss::future<> connection_context::process_one_request() {
         co_return;
     }
     _server.handler_probe(h->key).add_bytes_received(sz.value());
+    /**
+     * An entry point for the MPX serverless extensions. If the first request
+     * for a given connection has a special client_id then MPX extensions are
+     * enabled for this physical connection.
+     */
+    if (_server.enable_mpx_extensions()) {
+        if (unlikely(
+              is_first_request()
+              && h->client_id == multi_proxy_initial_client_id)) {
```

Comment:
so what happens if the first request doesn't have this magic client id, and then later the connection starts behaving like its virtualized?

Reply:
no, in this case connection is not virtualized